packages/platform-browser/src/hydration.ts

```diff
@@ -93,6 +99,22 @@ export function withNoHttpTransferCache():
   return hydrationFeature(HydrationFeatureKind.NoHttpTransferCache);
 }
 
+/**
+ * The function accepts a an object, which allows to configure cache parameters,
+ * such as which headers should be included (no headers are included by default),
+ * wether POST requests should be cached or a callback function to determine if a
+ * particular request should be cached.
+ *
+ * @publicApi
+ * @developerPreview
+ */
+export function withHttpTransferCacheOptions(
+    options: HttpTransferCacheOptions,
+    ): HydrationFeature<HydrationFeatureKind.HttpTransferCacheOptions> {
+  // This feature has no providers and acts as a flag to pass options to the HTTP transfer cache.
+  return hydrationFeature(HydrationFeatureKind.HttpTransferCacheOptions, [], options);
```

Comment:
Can we use `ɵwithHttpTransferCache` here directly, so that we have a set of providers and don't need to introduce a new field on the feature interface?
```suggestion
  return hydrationFeature(HydrationFeatureKind.HttpTransferCacheOptions, ɵwithHttpTransferCache(options).ɵproviders);
```